packages/components/src/ComboBox/src/ComboBox.tsx

```diff
@@ -0,0 +1,337 @@
+import { ClearContainerSlots, ClearProviders, composeClassnameRenderProps, cssModule, isTextOnlyChildren, SlotProvider } from "../../utils/index.ts";
+
+import { ComboBoxContext } from "./ComboBoxContext.ts";
+
+import styles from "./ComboBox.module.css";
+
+export const GlobalComboBoxCssSelector = "hop-ComboBox";
+
+export type ComboBoxSize = "sm" | "md";
+export type ComboBoxTriggerProps = StyledComponentProps<RACGroupProps>;
+
+export interface ComboBoxProps<T extends object> extends StyledComponentProps<Omit<RACComboBoxProps<T>, "children">> {
+    /**
+     * The items of the ComboBox.
+     */
+    children: ReactNode | ((item: T) => ReactNode);
+    /**
+     * The field children of the ComboBox. This includes the field label, field description, and field errors.
+     */
+    fieldChildren?: ReactNode;
+    /**
+     * The footer of the ComboBox.
+     */
+    footer?: ReactNode;
+    /**
+     * A ref for the HTML input element.
+     */
+    inputRef?: MutableRefObject<HTMLInputElement>;
+    /**
+     * If `true`, the select menu will not be the width of the trigger and instead be the width of its contents.
+     */
+    isAutoMenuWidth?: boolean;
+    /**
+     * If `true`, the select will take all available width.
+     * @default false
+     */
+    isFluid?: ResponsiveProp<boolean>;
+    /**
+     * Item objects in the collection.
+     */
+    items?: Iterable<T>;
+    /**
+     * The list box props.
+     */
+    listBoxProps?: ListBoxProps<T>;
+    /**
+     * The placeholder text when the select is empty.
+     */
+    placeholder?: string;
+    /**
+     * The popover props.
+     */
+    popoverProps?: PopoverProps;
+    /**
+     * An icon or text to display at the start of the select trigger.
+     */
+    prefix?: ReactNode;
+    /**
+     * The size of the select.
+     * @default "sm"
+     */
+    size?: ResponsiveProp<ComboBoxSize>;
+    /**
+     * The props for the select's trigger.
+     */
+    triggerProps?: ComboBoxTriggerProps;
+}
+
+function ComboBox<T extends object>(props: ComboBoxProps<T>, ref: ForwardedRef<HTMLDivElement>) {
+    // we extract the inputRef props, since we want to manually merge it with the context props.
+    const {
+        inputRef: userProvidedInputRef = null,
+        ...propsWithoutRef
+    } = props;
+    [props, ref] = useContextProps(propsWithoutRef, ref, ComboBoxContext);
+    const { stylingProps, ...ownProps } = useStyledSystem(props);
+    const {
+        className,
+        children,
+        fieldChildren,
+        footer,
+        isAutoMenuWidth,
+        isFluid: isFluidProp,
+        isInvalid,
+        items,
+        menuTrigger = "focus",
+        listBoxProps,
+        placeholder,
+        popoverProps,
+        prefix,
+        size: sizeProp,
+        style: styleProp,
+        triggerProps,
+        ...otherProps
+    } = ownProps;
+    const inputRef = useObjectRef(mergeRefs(userProvidedInputRef, props.inputRef !== undefined ? props.inputRef : null));
```

Comment:
```suggestion
    const inputRef = useObjectRef(mergeRefs(userProvidedInputRef, props.inputRef ?? null));
```

Reply:
hmm this was copied code from elsewhere. wonder why it wasn't done like this. any way it can backfire?

Reply:
Not that I can think of right now